src/coreclr/tools/superpmi/superpmi-shim-collector/jithost.cpp

```diff
@@ -26,26 +26,75 @@ JitHost* g_ourJitHost;
 // in the method context.
 bool RecordVariable(const WCHAR* key)
 {
-    // Special-case: we don't want to store some DOTNET variables during
-    // collections that we don't want to see on replay:
-    //   DOTNET_JitName -- used to get the VM to load the SuperPMI collection shim
-    //                      without requiring the shim to overwrite the original JIT.
-    //                      This JIT doesn't care about this on SuperPMI replay, but
-    //                      we don't need to waste the space in the MC file storing it.
-    //   DOTNET_AltJitName -- if collecting with an altjit, this is set. The JIT doesn't
-    //                      use this on replay, but it doesn't need to be stored.
-    //   DOTNET_EnableExtraSuperPmiQueries -- used to force the JIT to ask additional
-    //                      questions during SuperPMI collection. We don't want to store
-    //                      this variable because we don't want to replay using it.
-
-    if ((_wcsicmp(key, W("JitName")) == 0) ||
-        (_wcsicmp(key, W("AltJitName")) == 0) ||
-        (_wcsicmp(key, W("EnableExtraSuperPmiQueries")) == 0))
+    // Special cases: we don't want to store some DOTNET variables during
+    // collections, typically when they refer to file paths or simply because
+    // it does not make sense to replay with it.
+
```

Comment:
I verified that `DOTNET_JitName` and `DOTNET_AltJitName` do not come through this path since the JIT doesn't query them. So they don't need to be here.